scipy/optimize/least_squares.py

```diff
@@ -0,0 +1,818 @@
+"""Generic interface for least-square minimization."""
+
+from warnings import warn
+
+import numpy as np
+from numpy.linalg import norm
+
+from . import _minpack
+from .optimize import OptimizeResult
+from ._numdiff import approx_derivative, group_columns
+from ..sparse import issparse, csr_matrix
+from ..sparse.linalg import LinearOperator
+from ._lsq_common import (EPS, IMPLEMENTED_LOSSES, in_bounds,
+                          compute_loss_and_derivatives, compute_cost)
+from ._lsq_trf import trf
+from ._lsq_dogbox import dogbox
```

Comment:
All these `from ._lsq_XXX import` make me wonder if a `_least_square` package with all these modules inside along with this one as `__ini__.py` would not be better.